src/frontegg-auth/src/client.rs

```diff
@@ -36,35 +31,19 @@ impl Default for Client {
 impl Client {
     /// The environmentd Client. Do not change these without a review from the surfaces team.
     pub fn environmentd_default() -> Self {
-        Self {
-            client: reqwest::Client::builder()
-                .timeout(Duration::from_secs(5))
-                .build()
-                .expect("must build Client"),
-            retry_backoff: Duration::from_secs(1),
-            max_retry: Duration::from_secs(30),
-        }
-    }
-
-    async fn request<T, F, U>(&self, f: F) -> Result<T, Error>
-    where
-        F: Copy + Fn(reqwest::Client) -> U,
-        U: Future<Output = Result<T, Error>>,
-    {
-        Retry::default()
-            .clamp_backoff(self.retry_backoff.clone())
-            .max_duration(self.max_retry.clone())
-            .retry_async_canceling(|state| async move {
-                // Return a Result<Result<T, Error>, Error> so we can
-                // differentiate a retryable or not error.
-                match f(self.client.clone()).await {
-                    Err(Error::ReqwestError(err)) if err.is_timeout() => {
-                        warn!("frontegg timeout, attempt {}: {}", state.i, err);
-                        Err(Error::from(err))
-                    }
-                    v => Ok(v),
-                }
-            })
-            .await?
+        let retry_policy = ExponentialBackoff::builder()
```

Comment:
If this fixes the issue, it must mean that this retry_policy also knows that it should retry 500s? What's the thing making that decision? Why wasn't that happening before if we already had retries? Can you add a test for this situation? We have a frontegg mock test, and getting it to produce a 500 should be straightforward to add in.

Reply:
The retry_policy defines _when_ to retry, so it replaces the `mz_ore::retry::Retry` helper. _What_ gets retried is defined by the [library](https://github.com/TrueLayer/reqwest-middleware/blob/main/reqwest-retry/src/retryable_strategy.rs#L113).

I added a test that starts a local server and returns a given status code 2 times before returning a success, and then assert that certain response codes get retried. Let me know if I should add more!